storage/reads/array_cursor.gen.go

```diff
@@ -200,90 +201,122 @@ func (c floatArraySumCursor) Next() *cursors.FloatArray {
 	}
 }
 
-type integerFloatCountArrayCursor struct {
+type integerFloatWindowCountArrayCursor struct {
 	cursors.FloatArrayCursor
+	every int64
+	res   *cursors.IntegerArray
+	tmp   *cursors.FloatArray
 }
 
-func (c *integerFloatCountArrayCursor) Stats() cursors.CursorStats {
-	return c.FloatArrayCursor.Stats()
-}
-
-func (c *integerFloatCountArrayCursor) Next() *cursors.IntegerArray {
-	a := c.FloatArrayCursor.Next()
-	if len(a.Timestamps) == 0 {
-		return &cursors.IntegerArray{}
+func newIntegerFloatWindowCountArrayCursor(cur cursors.FloatArrayCursor, every int64) *integerFloatWindowCountArrayCursor {
+	resLen := MaxPointsPerBlock
+	if every == 0 {
+		resLen = 1
 	}
-
-	ts := a.Timestamps[0]
-	var acc int64
-	for {
-		acc += int64(len(a.Timestamps))
-		a = c.FloatArrayCursor.Next()
-		if len(a.Timestamps) == 0 {
-			res := cursors.NewIntegerArrayLen(1)
-			res.Timestamps[0] = ts
-			res.Values[0] = acc
-			return res
-		}
+	return &integerFloatWindowCountArrayCursor{
+		FloatArrayCursor: cur,
+		every:            every,
+		res:              cursors.NewIntegerArrayLen(resLen),
+		tmp:              &cursors.FloatArray{},
 	}
 }
 
-type integerFloatWindowCountArrayCursor struct {
-	cursors.FloatArrayCursor
-	every int64
+func newIntegerFloatCountArrayCursor(cur cursors.FloatArrayCursor) *integerFloatWindowCountArrayCursor {
+	// zero means aggregate over the whole series
+	return newIntegerFloatWindowCountArrayCursor(cur, 0)
 }
 
 func (c *integerFloatWindowCountArrayCursor) Stats() cursors.CursorStats {
 	return c.FloatArrayCursor.Stats()
 }
 
 func (c *integerFloatWindowCountArrayCursor) Next() *cursors.IntegerArray {
-	a := c.FloatArrayCursor.Next()
+	pos := 0
+	c.res.Timestamps = c.res.Timestamps[:cap(c.res.Timestamps)]
+	c.res.Values = c.res.Values[:cap(c.res.Values)]
+
+	var a *cursors.FloatArray
+	if c.tmp.Len() > 0 {
+		a = c.tmp
+	} else {
+		a = c.FloatArrayCursor.Next()
+	}
+
 	if a.Len() == 0 {
 		return &cursors.IntegerArray{}
 	}
 
-	res := cursors.NewIntegerArrayLen(0)
 	rowIdx := 0
 	var acc int64 = 0
 
-	firstTimestamp := a.Timestamps[rowIdx]
-	windowStart := firstTimestamp - firstTimestamp%c.every
-	windowEnd := windowStart + c.every
+	var windowEnd int64
+	if c.every != 0 {
```

Comment:
I think you can just use every = INT_MAX in `newIntegerFloatCountArrayCursor`, and eliminate the if here.
If every == INT_MAX, windowStart = 0, windowEnd = 0 + INT_MAX = INT_MAX

Reply:
How do you figure that `windowEnd` would be `INT_MAX`? `windowStart` is not likely to be zero.